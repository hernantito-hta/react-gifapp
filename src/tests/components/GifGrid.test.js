import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Testing component <GifGrid />', () => {
    const category = 'Science';
    test('should render the component correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        });
        let wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('should validate useFetchGifs', () => {
        const gifs = [
            {
                id: 'abc',
                title: 'Gost Boosters',
                url: 'https://localhost/any/image.jpg'
            },
            {
                id: '123',
                title: 'Gost Boosters',
                url: 'https://localhost/any/image.jpg'
            }
        ]
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        let wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
    
    
})
