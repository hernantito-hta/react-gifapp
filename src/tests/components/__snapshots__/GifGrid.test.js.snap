// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component <GifGrid /> should render the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Science
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing component <GifGrid /> should validate useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Science
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Gost Boosters"
      url="https://localhost/any/image.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Gost Boosters"
      url="https://localhost/any/image.jpg"
    />
  </div>
</Fragment>
`;
